### PR TITLE
Fix ToAsync for Task<Either<L,R>>

### PR DESCRIPTION
**Pull request by YTsalko**
Repo steps:

```csharp
TryAsync( async () => {
    throw new NotImplementedException();
    return 1; 
})
.ToEither()
.ToAsync()
.Match(
    v => Console.WriteLine($"In right - {v}"),
    e => Console.WriteLine("In left")
);
```

Expected result:
`In left`

Actual result:
`In right - 0`

### CODE REVIEW

**Comment by bender2k14** [comment]
Very nice.  Can you also add a test that covers this case?

**Comment by YTsalko** [comment]
I don't what to damage repo tests structure, so I'm not sure where to put this test. To EitherAsyncTests or create new file?

**Comment by bender2k14** [comment]
Yea, you can add the test to `EitherAsyncTests`.  It is funny actually that the file is currently empty.

**Comment by YTsalko** [comment]
added test

File: LanguageExt.Tests/EitherAsyncTests.cs [hunk 1]
```diff
@@ -2,10 +2,23 @@
...
     {
+        [Fact]
+        public async Task ToAsyncExtensionLeftTest()
+        {
+            var eitherAsync = Left<Exception, Unit>(new Exception()).AsTask().ToAsync();
+
+            await eitherAsync.Match(
+                v => Assert.True(false), // should never happen
+                e => Assert.True(true)
+            );
```
[the remark this answers is not in the document]
Comment (YTsalko) [reply LanguageExt.Tests/EitherAsyncTests.cs]: done. Thanks.

**Comment by louthy** [comment]
Thanks 👍

**Comment by bender2k14** [comment]
> If it's going to be a while, what's the workaround to using .ToAsync()?

In the meantime, you can just implement your own extension method called `BugFreeToAsync` with the same implementation as the yet-to-be released code.

**Comment by NightWatchman** [comment]
Thanks for the info. I gave that a shot a little while ago, but I think I ran into an issue with the EitherData constructor being protected and inaccessible from my code. I need to revisit this to verify that was the issue or if it was something else, but it wasn't working for me because of something like that.

**Comment by bender2k14** [comment]
Yea, I see what you mean @NightWatchman.  I couldn't figure out how to use the same implementation either.  Sorry for misleading you.

Here is a workaround for you though.

```
public static EitherAsync<L, R> BugFreeToAsync<L, R>(this Task<Either<L, R>> mma) => mma
  .Map(ma => ma.ToAsync())
  .Apply(mmma => default(MEitherAsync<L, R>).RunAsync(_ => mmma));
```

To be sure, you can check as I have that it passes these tests.
```
[Fact]
public async Task TaskEitherBugFreeToAsync_Left_IsLeft()
{
    var actual = await Left(0)
        .Bind<string>()
        .AsTask()
        .BugFreeToAsync()
        .IsLeft;
    Assert.True(actual);
}

[Fact]
public async Task TaskEitherBugFreeToAsync_Right_IsRight()
{
    var actual = await Right(0)
        .Bind<string>()
        .AsTask()
        .BugFreeToAsync()
        .IsRight;
    Assert.True(actual);
}
```

**Comment by louthy** [comment]
Sorry, I thought I’d deployed it already.  I’m away for a couple of days.  I’ll try and sort it this weekend

**Pull request merged**